internal/provider/data_source_hvn_route.go

```diff
@@ -0,0 +1,104 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func dataSourceHVNRoute() *schema.Resource {
+	return &schema.Resource{
+		Description: "The HVN Route data source provides information about an existing HVN route.",
+		ReadContext: dataSourceHVNRouteRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &hvnRouteDefaultTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn": {
+				Description: "The `self_link` of the HashiCorp Virtual Network (HVN).",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"destination_cidr": {
+				Description: "The destination CIDR of the HVN route",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			// Computed outputs
+			"self_link": {
+				Description: "A unique URL identifying the HVN route.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"target": {
+				Description: "The target of the HVN route.",
+				Type:        schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed: true,
+			},
+			"state": {
+				Description: "The state of the HVN route.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "The time that the HVN route was created.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceHVNRouteRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+
+	hvn := d.Get("hvn").(string)
+	var hvnLink *sharedmodels.HashicorpCloudLocationLink
+
+	hvnLink, err := parseLinkURL(hvn, "hashicorp.network.hvn")
```

Comment:
We have a `HvnResourceType` constant that you can use here instead of the raw string